### PR TITLE
Port subobject changes to dynamic typebindings

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Ported changes to dynamic branch.

#### Tests
Ran test suite.
Also ran my local ability system tests.

#### Primary reviewers
@Vatyx @improbable-valentyn

### CODE REVIEW

File: Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -409,6 +409,11 @@ bool USpatialActorChannel::ReplicateActor()
 
 bool USpatialActorChannel::ReplicateSubobject(UObject *Object, const FReplicationFlags &RepFlags)
 {
+	if (NetDriver->TypebindingManager->IsSupportedClass(Object->GetClass()) == false)
```
Comment (improbable-valentyn) [inline Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: Nit: prefer `if (!NetDriver->TypebindingManager->IsSupportedClass(Object->GetClass()))`

File: Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp [hunk 2]
```diff
@@ -129,11 +129,22 @@ void USpatialTypebindingManager::CreateTypebindings()
...
-								Info.ComponentClasses.Add(Node->ComponentTemplate->GetClass());
+								Info.SubobjectClasses.Add(Node->ComponentTemplate->GetClass());
 							}
 						}
 					}
 				}
+
+				// Iterate over all subobjects and add any non ActorComponents
+				TArray<UObject*> DefaultSubobjects;
+				ContainerCDO->GetDefaultSubobjects(DefaultSubobjects);
```
[the remark this answers is not in the document]
Comment (m-samiec) [reply Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp]: No should be fine. Default objects aren't added to the CDO, but they should exist on the actor itself.

[the remark this answers is not in the document]
Comment (m-samiec) [reply Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp]: Oh wait, sorry, my context here was wrong. Yes, if it's a blueprint subobject, it won't find it.

File: Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 3]
```diff
@@ -649,18 +649,20 @@ UObject* USpatialReceiver::GetTargetObjectFromChannelAndClass(USpatialActorChann
...
-	else
-	{
-		checkNoEntry();
+		check(ActorInfo->SubobjectClasses.Find(Class));
+
+		TArray<UObject*> DefaultSubobjects;
+		Channel->Actor->GetDefaultSubobjects(DefaultSubobjects);
+		UObject** FoundSubobject = DefaultSubobjects.FindByPredicate([Class](const UObject* Obj)
+		{
+			return (Obj->GetClass() == Class);
```
Comment (improbable-valentyn) [inline Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: `Obj->IsA(Class)`

File: Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp [hunk 4]
```diff
@@ -129,11 +129,22 @@ void USpatialTypebindingManager::CreateTypebindings()
...
+				for (auto Subobject : DefaultSubobjects)
+				{
+					if (Cast<UActorComponent>(Subobject) == nullptr)
```
Comment (improbable-valentyn) [inline Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp]: `if (!Subobject->IsA<UActorComponent>())`

**Pull request merged**